File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/SetPersonPropertiesOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.posthog.PosthogHelper;
import java.util.Map;
import org.json.JSONException;

public class SetPersonPropertiesOptions {

    @Nullable
    private final Map<String, Object> properties;

    @Nullable
    private final Map<String, Object> setOnceProperties;

    public SetPersonPropertiesOptions(@NonNull PluginCall call) throws JSONException {
        this.properties = PosthogHelper.createHashMapFromJSONObject(call.getObject("properties"));
        this.setOnceProperties = PosthogHelper.createHashMapFromJSONObject(call.getObject("setOnceProperties"));
    }

    @Nullable
    public Map<String, Object> getProperties() {
        return properties;
    }

    @Nullable
    public Map<String, Object> getSetOnceProperties() {
        return setOnceProperties;
    }
}

File: packages/posthog/ios/Plugin/Classes/Options/SetPersonPropertiesOptions.swift
import Foundation
import Capacitor

@objc public class SetPersonPropertiesOptions: NSObject {
    private var properties: [String: Any]?
    private var setOnceProperties: [String: Any]?

    init(call: CAPPluginCall) {
        self.properties = PosthogHelper.createHashMapFromJSObject(call.getObject("properties"))
        self.setOnceProperties = PosthogHelper.createHashMapFromJSObject(call.getObject("setOnceProperties"))
    }

    func getProperties() -> [String: Any]? {
        return properties
    }

    func getSetOnceProperties() -> [String: Any]? {
        return setOnceProperties
    }
}
